forgerock-authenticator/src/main/java/org/forgerock/android/auth/AuthenticatorManager.java

```diff
@@ -240,9 +289,69 @@ List<PushNotification> getAllNotifications(@NonNull Mechanism mechanism) {
         }
     }
 
+    private void createCombinedMechanismsFromUri(String uri, FRAListener<Mechanism> listener) {
+        final Mechanism[] oathMechanism = new Mechanism[1];
+
+        // Check if security policies are available in the URI before proceed with registration
+        Logger.debug(TAG, "Evaluating policies for the new Account");
+        boolean policiesCompliance = policyEvaluator.evaluate(context, uri);
+        if (!policiesCompliance) {
+            listener.onException(new MechanismCreationException("This account cannot be " +
+                    "registered on this device. It violates the following policy: " +
+                    policyEvaluator.getNonCompliancePolicy().getName()));
+            return;
+        } else {
+            Logger.debug(TAG, "All policies passed for the new Account");
+        }
+
+        // Register OATH mechanism passing combined parameters
+        oathFactory.createFromUri(uri, new FRAListener<Mechanism>() {
+            @Override
+            public void onSuccess(Mechanism result) {
+                oathMechanism[0] = result;
+                Logger.debug(TAG, "OATH mechanism in Combined MFA successfully created.");
+            }
+
+            @Override
+            public void onException(Exception e) {
+                Logger.error(TAG, "Error creating OATH mechanism in Combined MFA.");
+            }
+        });
+
+        // Register PUSH mechanism passing combined parameters
+        pushFactory.createFromUri(uri, new FRAListener<Mechanism>() {
+            @Override
+            public void onSuccess(Mechanism result) {
+                listener.onSuccess(result);
```

Comment:
We just want to return one result?

Reply:
Yes, the idea is to return the last registered. Return both mechanisms as result would require refactor the listener to FRAListener<Mechanism[]>, which would require a huge refactor across the SDK and Authenticator sub products

Reply:
Well, this will be 4.0, so we maybe be able to return both we introduced a break change on FRAClient.createMechanismFromUri. I think it will be better

Reply:
The idea is to try best to register? if OATH failed, do we still want to proceed with the Push?

Reply:
Yes, that's what I got from Product. Previously, if anything fail, the registration was not completed.